docs/rules/no-context.md

```diff
@@ -0,0 +1,51 @@
+# Prevent usage of context (no-context)
+
+[Context](https://facebook.github.io/react/docs/context.html) is an advanced and experimental feature. The API is likely to change in future releases. Most applications will never need to use context. Especially if you are just getting started with React, you likely do not want to use context. Using context will make your code harder to understand because it makes the data flow less clear. It is similar to using global variables to pass state through your application.
```

Comment:
It would be nice to point people toward things like higher order components if they want to use context.